src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -73,4 +65,114 @@ private void validDistance(Long newDistance, Long exitDistance) {
     private void addSection(Section section) {
         sections.add(section);
     }
+
+    public List<Station> getStations() {
+        if (sections.isEmpty()) {
+            return Arrays.asList();
+        }
+
+        return createStations();
+    }
+
+    private List<Station> createStations() {
+        List<Station> stations = new ArrayList<>();
+
+        Section section = getFirstSection();
+        stations.add(section.getUpStation());
+        stations.add(section.getDownStation());
+
+        findStations(stations, section);
+        return stations;
+    }
+
+    private Section getFirstSection() {
+        Station station = findUpStation();
+
+        return sections.stream()
+                .filter(section -> section.getUpStation().equals(station))
+                .findFirst()
+                .orElseThrow(() -> new NoSuchElementException("노선의 역을 찾을 수 없습니다."));
+    }
+
+    private Station findUpStation() {
+        List<Station> upStations = this.sections.stream()
+                .map(Section::getUpStation)
+                .collect(Collectors.toList());
+
+        return upStations.stream()
+                .filter(station -> isNotContainsDownStation(station))
+                .findFirst()
+                .orElseThrow(() -> new NoSuchElementException("노선의 역을 찾을 수 없습니다."));
+    }
+
+    private boolean isNotContainsDownStation(Station station) {
+        return sections.stream()
+                .noneMatch(section -> section.getDownStation().equals(station));
```

Comment:
getter 를 사용하기 보다는 객체에 메세지를 보내보면 어떨까요? 😄 
```suggestion
        return sections.stream()
                .noneMatch(section -> section.isMatchedDownStation(station));
```

Reply:
생각보다 `section.getDownStation().equals(station)` 쓰는 곳이 많았네요ㅠㅠ
피드백 주신 것처럼 개선 완료하였습니다!